source/faq.txt

```diff
@@ -4,12 +4,176 @@ FAQ
 
 .. default-domain:: mongodb
 
-Frequently Asked Questions
-
 .. contents:: On this page
    :local:
    :backlinks: none
-   :depth: 1
+   :depth: 2
    :class: singlecol
 
-Lorem ipsum.
+General
+-------
+
+Why are there two versions of the classes? e.g. ``Mongo`` and ``MongoClient``?
```

Comment:
The 4.x driver does not have a `Mongo` class anymore.  Rather, there are two (not including reactive) `MongoClient` types.

The first, `com.mongodb.MongoClient`, is a class that has been in the driver for a long time.  It supports both the "modern" API, which includes

* Configuration with `MongoClientSettings` and `ConnectionString`
* CRUD API via `MongoDatabase`, and from there, `MongoCollection`

It also supports the "legacy" API, which includes:

* Configuration with `MongoClientOptions` and `MongoClientURI`
* CRUD API via `DB`, and from there, `DBCollection`

Instances of this class are created via its constructors

The second, `com.mongodb.client.MongoClient`, is an interface introduced in the 3.7 release.  It supports only the "modern" API (as described above)

Instances of this type are create via factory methods defined in the `com.mongodb.client.MongoClients` class.

As to why we added the new interface and configuration

1. We wanted a cleaner API for new users that didn't have the confusion of including multiple CRUD APIs
2. We wanted to ensure that the new CRUD API was available in a Java package structure that would work well with Java module support introduced in Java 9.

Reply:
I don't think it's important to capture they why factor here, so I've reworded the question to: "Are there two versions of MongoClient" to capture the information you've provided.